python/src/lib.rs

```diff
@@ -32,6 +32,40 @@ impl Level {
     }
 }
 
+/// This struct is used to keep track of the cursor of a cluster scan.
+/// We want to avoid passing the cursor between layers of the application,
+/// So we keep the state in the container and only pass the id of the cursor.
+/// The cursor is stored in the container and can be retrieved using the id.
+/// The cursor is removed from the container when the object is deleted (dropped).
+#[pyclass]
+#[derive(Default)]
+pub struct ClusterScanCursor {
+    cursor: String,
+}
+
+#[pymethods]
+impl ClusterScanCursor {
+    #[new]
+    fn new(new_cursor: Option<String>) -> Self {
+        match new_cursor {
+            Some(cursor) => ClusterScanCursor { cursor },
+            None => ClusterScanCursor::default(),
+        }
+    }
+
+    fn get_cursor(&self) -> String {
+        self.cursor.clone()
+    }
+
+    fn __del__(&mut self) {
+        glide_core::cluster_scan_container::remove_scan_state_cursor(self.cursor.clone());
```

Comment:
If the cursor is "finished", this should not do anything right?

Reply:
Yep, if the cursor is empty, there's nothing in the table so removal will do nothing.